aspnetcore/tutorials/choose-web-ui.md

```diff
@@ -0,0 +1,193 @@
+
+Blazor Server benefits:
+
+* Share app logic across server and client..
+* Flexibility of direct access to server resources. For example:
+  * Database.
+  * Secrets, such as access key-values used for API calls to Azure storage.
+* Uses a token-based security model, rather than using cookies. For example, cookies may not be enabled on the browser.
+
+To get started with ASP.NET Core Blazor Server, see [Get started with Blazor](https://dotnet.microsoft.com/learn/aspnet/blazor-tutorial/intro). For an overview of ASP.NET Core Blazor Server, its architecture and benefits, see [Introduction to ASP.NET Core Blazor](xref:blazor/index).
+
+### ASP.NET Core MVC plus hosted Blazor WebAssembly
+
+MVC and Blazor are both part of the ASP.NET Core framework and are designed to be used together. Razor components can be integrated into Razor Pages and MVC apps in a hosted Blazor WebAssembly solution. When the page or view is rendered, components can be pre-rendered at the same time.
+
+Choose ASP.NET Core MVC plus Blazor WebAssembly when these benefits best fit your Web UI app needs:
+
+* Prerendering:
+  * Executes Blazor components on the server and renders them into a page or view.
+	* Improves the perceived load time of the app while interactivity is being set up.
+* Add islands of interactivity to existing views (pages), using the component tag helper.
+
+To get started with ASP.NET Core MVC plus Blazor WebAssembly, see [Prerender and integrate ASP.NET Core Razor components](xref:blazor/components/prerendering-and-integration).
+
+### ASP.NET Core Single Page Application (SPA) with JavaScript Frameworks such as Angular and React
+
+Build client-side logic for ASP.NET Core apps using popular JavaScript frameworks, like [Angular](https://angular.io/) or [React](https://facebook.github.io/react/). ASP.NET Core provides project templates for Angular and React.
+
+Benefits of ASP.NET Core SPA with JavaScript Frameworks, in addition to the client rendering benefits previously listed:
+
+* The JavaScript runtime environment is already provided with the browser.
+* Large community and mature ecosystem.
+* Build client-side logic for ASP.NET Core apps using popular JS frameworks, like Angular and React, or build your client-side logic with .NET using Blazor.
+
+Downsides:
+* More coding languages, frameworks, and tools required.
+* Difficult to share code so some logic may be duplicated.
+
+To get started, see:
+
+* [Use the Angular project template with ASP.NET Core](xref:spa/angular)
+* [Use the React project template with ASP.NET Core](xref:spa/react)
+
+## Next steps
+
+For more information, see:
+
+* [Blazor hosting models: Server vs WebAssembly](xref:blazor/hosting-models)
+* [Provide an API for your web UI app: Compare gRPC services with HTTP APIS and SignalR](xref:grpc/comparison)
```

Comment:
The title for this, *Compare gRPC services with HTTP APIs*, seems ok to **_Auto-Suck_**&trade; ...... "Auto-Suck" :smile: ... I can see it's going to *one of those days here*. :smile: lol

```suggestion
* <xref:grpc/comparison>
```